clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java

```diff
@@ -193,7 +199,8 @@ public OffsetCommitRequest build(short version) {
                 case 1:
                 case 2:
                 case 3:
-                    long retentionTime = version == 1 ? DEFAULT_RETENTION_TIME : this.retentionTime;
+                case 4:
+                    long retentionTime = version == 1 || version == 4 ? DEFAULT_RETENTION_TIME : this.retentionTime;
```

Comment:
We have this ability to set the retention time, but we don't really use it anywhere in the code other than test cases. Since we're not planning on supporting it going forward, can we just remove the `retentionTime` field and just hard-code the use of `DEFAULT_RETENTION_TIME` for older requests?

Reply:
I'm not sure I understand your point. Shouldn't older request (v2 or v3) still be able to provide a customized retention time? Wouldn't removing the field prevent that?

Reply:
My point is that we haven't actually exposed that capability. The consumer always uses the default value. Since we don't need it going forward, we can just simplify things internally.

Reply:
Sounds good. I have removed this field in the new commit. Please check if the change covers what you have in mind. Thanks!